core/src/OpenACC/Kokkos_OpenACC_Instance.hpp

```diff
@@ -46,24 +46,31 @@
 #define KOKKOS_OPENACC_INSTANCE_HPP
 
 #include <impl/Kokkos_InitializationSettings.hpp>
+#include <impl/Kokkos_Tools.hpp>
 
 #include <cstdint>
 #include <iosfwd>
 #include <string>
+#include <openacc.h>
 
 namespace Kokkos::Experimental::Impl {
 
 class OpenACCInternal {
   bool m_is_initialized = false;
 
-  OpenACCInternal()                       = default;
   OpenACCInternal(const OpenACCInternal&) = default;
   OpenACCInternal& operator=(const OpenACCInternal&) = default;
 
  public:
-  static OpenACCInternal* singleton();
+  int m_accDev;
+  int m_async_id;
+  uint32_t m_instance_id;
 
-  void initialize(InitializationSettings const& settings);
+  static OpenACCInternal& singleton();
+
+  int verify_is_initialized(const char* const label) const;
+
+  void initialize(int device_id, int async_arg = acc_async_sync);
```

Comment:
I don't think passing a device ID is a good idea considering that it is only legal to use the device selected at initialization

Reply:
The `device_id` parameter is removed.